simpeg/optimization.py

```diff
@@ -851,9 +874,120 @@ def bindingSet(self, x):
         Optimality condition. (Satisfies Kuhn-Tucker) MoreToraldo91
 
         """
-        bind_up = np.logical_and(x == self.lower, self.g >= 0)
-        bind_low = np.logical_and(x == self.upper, self.g <= 0)
-        return np.logical_or(bind_up, bind_low)
+        out = np.zeros(x.shape, dtype=bool)
+        if self.lower is not None:
+            out |= (x <= self.lower) & (self.g >= 0)
+        if self.upper is not None:
+            out |= (x >= self.upper) & (self.g <= 0)
+        return out
+
+
+class InexactCG(object):
```

Comment:
Used to hold the CG options... used by `ProjectedGradient`, `InexactGaussNewton` and `ProjectedGNCG` minimizers.